File: src/main/java/com/leijendary/spring/microservicetemplate/document/LocaleDocument.java
package com.leijendary.spring.microservicetemplate.document;

import lombok.Data;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

import static org.springframework.data.elasticsearch.annotations.FieldType.Keyword;

@Data
public abstract class LocaleDocument {

    @Field(type = Keyword)
    private String language;

    @Field(type = FieldType.Integer)
    private int ordinal;
}

File: src/main/kotlin/com/leijendary/spring/core/document/LocaleDocument.kt
package com.leijendary.spring.core.document

import com.leijendary.spring.core.projection.LocaleProjection
import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType.Integer
import org.springframework.data.elasticsearch.annotations.FieldType.Keyword

open class LocaleDocument(
    @Field(type = Keyword)
    override var language: String,

    @Field(type = Integer)
    override var ordinal: Int = 0
) : LocaleProjection
